src/Compilers/CSharp/Portable/Symbols/ReferenceManager.cs

```diff
@@ -567,36 +568,41 @@ private bool CreateAndSetSourceAssemblyFullBind(CSharpCompilation compilation)
 
                 foreach (int i in newSymbols)
                 {
+                    ref BoundInputAssembly currentBindingResult = ref bindingResult[i];
```

Comment:
This is another theme in this PR: stop passing around an array an index. The lack of array index state tracking makes this hard to manage. Instead just pass around a `ref` to the array element. This also has the benefit of reducing the amount of times we dereference the array at runtime.